Orange/widgets/evaluate/owtestlearners.py

```diff
@@ -222,16 +222,17 @@ def set_learner(self, learner, key):
         if key in self.learners and learner is None:
             # Removed
             del self.learners[key]
+            self._update_stats_model()
         else:
             self.learners[key] = Input(learner, None, None)
-
-        self._update_stats_model()
+            self._invalidate([key])
 
     def set_train_data(self, data):
         """
         Set the input training dataset.
         """
-        self.error(0)
+        self.error([0, 1, 2, 3])
```

Comment:
This clears all errors and warnings even if they are still relevant.
E.g. test data has no class so error 1 is shown, then train data is updated and the error disappears (but should still be shown)